src/blocks/gallery-carousel/context.js

```diff
@@ -0,0 +1,41 @@
+import { createContext, useState } from '@wordpress/element';
+
+const GalleryCarouselContext = createContext( {
+	setSelectedImage: () => {},
+	setCaptionFocused: () => {},
+	selectedImage: null,
+	captionFocused: false,
+	images: [],
+	isSelected: false,
+	showThumbnails: false,
+	gutter: 0,
+	gutterMobile: 0,
+} );
+
+const GalleryContextProvider = ( { gutter, gutterMobile, showThumbnails, children, images, isSelected } ) => {
+	const [ selectedImage, setSelectedImage ] = useState( null );
+	const [ captionFocused, setCaptionFocused ] = useState( false );
+
+	const galleryContext = {
+		selectedImage,
```

Comment:
A-Z order

You should run `yarn test:js -- src/blocks/gallery-carousel` and fix all the warnings